quantum/impl/quantum_mutex_impl.h

```diff
@@ -115,9 +119,8 @@ inline
 Mutex::Guard::Guard(Mutex& mutex,
                     Mutex::AdoptLock) :
     _mutex(&mutex),
-    _ownsLock(true)
+    _ownsLock(mutex.isLocked())
```

Comment:
Note for the future: we should probably have an assert here that verifies that the guard owner is the mutex owner; that requires the mutex itself to offer an ownsLock(TaskId = local::taskId()) API. Let's do that next time.